python/python_fastapi_server/routers/edr.py

```diff
@@ -391,7 +402,9 @@ def get_collectioninfo_for_id(
             crs=crs,
             output_formats=output_formats,
         )
-    return collection
+
+    get_cap = get_capabilities(edr_collection)
+    return collection, get_cap["expires"]
```

Comment:
We do a `get_capabilities` just to get an `expires`? Why not get this from the calls that are use to construct `collection`? 
Like `get_vertical_dim_for_collection`, `get_custom_dims_for_collection`, `get_times_for_collection`. `get_extent`, `get_ref_times_for_coll`...

I notice that this leads to 4 calls to GetCapabilities, so we are really reliant on caching for performance, even within a single call. I think it would be better if this were not the case, i.e. only a single call to GetCapabilities is done, and this is reused in the code, without relying on Redis or in-memory caching.

Besides that this doesn't feel right, I think an actual issue with the current setup is that data returned by `get_colection_info_for_id` could be based on inconsistent GetCapabilties responses from Adaguc, if the cache entry expires in the middle.

Reply:
One getCapabiltites call is now used for the entire collection information gathering